### PR TITLE
Add simple usage message to systemd-backlight

### PR DESCRIPTION
**Pull request by hubenchang0515**

### CODE REVIEW

**Comment by poettering** [comment]
hmm, this is typically not a tool mere mortals are supposed to call. That said, if people ask for it, we can certainly add a help text, but in that case it should follow our usual output, i.e. have a `help()` function, like for example src/notify/notify.c, and output roughly the same bits of information

**Comment by hubenchang0515** [comment]
How about this

File: src/backlight/backlight.c [hunk 1]
```diff
@@ -13,11 +13,37 @@
...
 
+static int help(void) {
+        _cleanup_free_ char *link = NULL;
+        int r;
+
+        r = terminal_urlify_man("systemd-backlight", "8", &link);
+        if (r < 0)
+                return log_oom();
+
+        printf("%s [save|load] [backlight|leds]:[DEVICE]\n"
```
Comment (poettering) [inline src/backlight/backlight.c]: we'd usually do this in two lines:

```
printf("%s save [backlight|leds]:DEVICE\n
        "%s load [backlight|leds]:DEVICE\n
...
```

File: src/backlight/backlight.c [hunk 2]
```diff
@@ -13,11 +13,37 @@
...
+               "\n%sSave and resume backlight brightness at shutdown and boot.%s\n\n"
```
Comment (poettering) [inline src/backlight/backlight.c]: if figure you mean resume → restore?

File: src/backlight/backlight.c [hunk 3]
```diff
@@ -13,11 +13,37 @@
...
+               "  save            Save current brightness\n"
+               "  load            Set brightness to be the previously saved value\n"
+               "  backlight       Operate backlight brightness\n"
+               "  leds            Operate LED brightness\n"
+               "  DEVICE          Device to be operated\n"
```
Comment (poettering) [inline src/backlight/backlight.c]: we explain verbs here only typically, i.e. these final three lines should go as people would assume they were supported verbs, but they aren't.

**Review by poettering — CHANGES_REQUESTED** [review]

File: src/backlight/backlight.c (hunk 3 above)
Comment (hubenchang0515) [reply src/backlight/backlight.c]: removed final three

File: src/backlight/backlight.c (hunk 2 above)
Comment (hubenchang0515) [reply src/backlight/backlight.c]: did

File: src/backlight/backlight.c (hunk 1 above)
Comment (hubenchang0515) [reply src/backlight/backlight.c]: did

File: src/backlight/backlight.c [hunk 4]
```diff
@@ -335,7 +360,7 @@ static int run(int argc, char *argv[]) {
         log_setup_service();
 
         if (argc != 3)
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "This program requires two arguments.");
+                return help();
```
Comment (keszybz) [inline src/backlight/backlight.c:363]: No, that is wrong. The program should print help only when `--help` is specified. When called with a wrong number of arguments, it should just print the message and return an error (as before).

File: src/backlight/backlight.c [hunk 5]
```diff
@@ -334,6 +359,9 @@ static int run(int argc, char *argv[]) {
 
         log_setup_service();
 
+        if (argc == 2 && streq(argv[1], "--help"))
```
Comment (keszybz) [inline src/backlight/backlight.c]: Normally we allow help to be present in any position, but I guess we can use this narrow check here: after all, we don't expect people to call the binary too often.

**Review by keszybz — COMMENTED** [review]
Any chance you could also add a description of the syntax to the man page? We usually try to keep `--help` output and the man page in sync.

**Comment by hubenchang0515** [comment]
Would you please teach me how to build and preview the man page

File: src/backlight/backlight.c (hunk 5 above)
Comment (keszybz) [reply src/backlight/backlight.c]: On second thought, please change this to:
```C
if (strv_contains(strv_skip(argv, 1), "--help")))
```
It's not much longer but it allows the param to be present in any place.

**Comment by keszybz** [comment]
> Would you please teach me how to build and preview the man page

`build/man/man systemd-backlight`
or
`build/man/html systemd-backlight`

File: src/backlight/backlight.c (hunk 5 above)
Comment (hubenchang0515) [reply src/backlight/backlight.c]: done

**Comment by keszybz** [comment]
Looks nice.

**Pull request merged**